set_alias.py

```diff
@@ -87,71 +151,61 @@ def slack_user_group(self):
                 self._slack_user_group = group
                 return group
 
-        raise ValueError('No user groups found that match {}'.format(self.slack_user_group_handle))
+        raise ValueError(f'No user groups found that match {self.slack_user_group_handle}')
 
-    @property
-    def on_call_email_addresses(self):
-        # todo rename oncall to in_rotation
+    def get_on_call_engs(self, weekday):
         """
-        Uses WEEKDAY_ROTATION on weekdays, otherwise uses the week number
+        Uses WEEKDAY_ROTA on weekdays, otherwise rotates the on-call in pairs
         based on the index stored in the last_oncall_index tag
         :return: All on-call email addresses
         """
-        if self._on_call_email_addresses is not None:
-            return self._on_call_email_addresses
+        users = set()
 
-        users = set()  # users can be in multiple schedule, this will de-dupe
+        if not settings.ROTATE_WEEKEND_ONCALL or weekday < 5:
+            oncalls = settings.WEEKDAY_ROTATION[weekday]
+        else:   # set a param that stores the index of the last on-call
+            eng_list = list(settings.People)
+            try:
+                oncall_index = int(SSM.get_parameter(Name='/oncall/last_oncall_index',
+                                                     WithDecryption=False)['Parameter']['Value'])
+            except ClientError:
+                oncall_index = 0
 
-        today = datetime.now(dateutil.tz.gettz(settings.TIMEZONE))
-        weekday = today.weekday()  # Monday is 0 and Sunday is 6
+            oncall_index = oncall_index % math.ceil(len(eng_list) / 2)
+            if weekday == 6:    # only increment once per weekend so we have the same pair
+                SSM.put_parameter(Name='/oncall/last_oncall_index',
```

Comment:
why do you want to store this? why not calculate based on week number, makes this simpler..

Reply:
i did use .week initially, but then realized it would have in some cases put the same pair oncall on consecutive weekends.  BUT i didn't know about isocalendar() so i'll update it to that